v03_pipeline/lib/tasks/reference_data/updated_reference_dataset_collection_task.py

```diff
@@ -0,0 +1,206 @@
+import os
+from datetime import datetime
+from functools import cached_property
+from types import FunctionType
+
+import hail as hl
+import luigi
+import pytz
+
+from v03_pipeline.lib.model import ReferenceDatasetCollection
+from v03_pipeline.lib.paths import valid_reference_dataset_collection_path
+from v03_pipeline.lib.reference_data.config import CONFIG
+from v03_pipeline.lib.tasks.base.base_update_task import BaseUpdateTask
+from v03_pipeline.lib.tasks.files import GCSorLocalTarget
+
+# NOTE: this is intended to replace the code in v03_pipeline/bin/write_combined_reference_ht.py, v03_pipeline/bin/write_combined_interval_reference_ht.py, and combine.py
+
+class UpdatedReferenceDatasetCollectionTask(BaseUpdateTask):
+    reference_dataset_collection = luigi.EnumParameter(enum=ReferenceDatasetCollection)
+    dataset = luigi.OptionalStrParameter()
+
+    @cached_property
+    def _destination_path(self) -> str:
+        return valid_reference_dataset_collection_path(
+            self.reference_genome,
+            self.dataset_type,
+            self.reference_dataset_collection,
+        )
+
+    def output(self) -> luigi.Target:
+        return GCSorLocalTarget(self._destination_path)
+
+    def initialize_table(self) -> hl.Table:
+        if self.dataset is None:
+            key_type = self.reference_dataset_collection.table_key_type(self.reference_genome)
+            return hl.Table.parallelize(
+                [],
+                key_type,
+                key=key_type.fields,
+                globals=hl.Struct(
+                    paths=hl.Struct(),
+                    versions=hl.Struct(),
+                    enums=hl.Struct(),
+                ),
+            )
+
+        if not hl.hadoop_exists(os.path.join(self._destination_path, '_SUCCESS')):
```

Comment:
I think you can probably do something like `GCSorLocalFolderTarget(self._destination_path).exists()` here.  It does literally the same thing as `hl.hadoop_exists` though... but while we're baked into `luigi` might as well standardize on the features.

Reply:
also... I _think_ even the above might be unnecessary, there's existing [logic](https://github.com/broadinstitute/seqr-loading-pipelines/blob/main/v03_pipeline/lib/tasks/base/base_update_task.py#L29-L33) in `BaseUpdateTask` that does this bit for us!

Reply:
oh yes I missed that logic, that simplifies my implementation of `initialize_table` for sure